pkgs/ffigen/test/native_objc_test/block_inherit_test.dart

```diff
@@ -0,0 +1,102 @@
+      DynamicLibrary.open(dylib.absolute.path);
+
+      generateBindingsForCoverage('block_inherit');
+    });
+
+    test('BlockInheritTestBase', () {
+      BlockInheritTestBase baseObj = BlockInheritTestBase.new1();
+      expect(baseObj.getAnimal().laysEggs(), false);
+      expect(baseObj.acceptAnimal_(Platypus.new1()), true);
+
+      ObjCBlock<Mammal Function()> returner = baseObj.getReturner();
+      Mammal returnerResult = returner();
+      expect(returnerResult.laysEggs(), false);
+
+      ObjCBlock<Bool Function(Platypus)> accepter = baseObj.getAccepter();
+      expect(accepter(Platypus.new1()), true);
+
+      final platypus = Platypus.new1();
+      ObjCBlock<Platypus Function()> platypusReturner =
+          ObjCBlock_Platypus.fromFunction(() => platypus);
+      expect(baseObj.invokeReturner_(platypusReturner).laysEggs(), true);
+
+      ObjCBlock<Bool Function(Mammal)> mammalAccepter =
+          ObjCBlock_bool_Mammal.fromFunction(
+              (Mammal mammal) => mammal.laysEggs());
+      expect(baseObj.invokeAccepter_(mammalAccepter), false);
+    });
+
+    test('BlockInheritTestChild', () {
+      BlockInheritTestChild childObj = BlockInheritTestChild.new1();
+      BlockInheritTestBase baseObj = childObj;
+      expect(baseObj.getAnimal().laysEggs(), true);
+      expect(baseObj.acceptAnimal_(Platypus.new1()), true);
+      expect(childObj.acceptAnimal_(Mammal.new1()), false);
+
+      ObjCBlock<Mammal Function()> baseReturner = baseObj.getReturner();
+      Mammal baseReturnerResult = baseReturner();
+      expect(baseReturnerResult.laysEggs(), true);
+
+      ObjCBlock<Platypus Function()> childReturner = childObj.getReturner();
+      Platypus childReturnerResult = childReturner();
+      expect(childReturnerResult.laysEggs(), true);
+
+      ObjCBlock<Bool Function(Platypus)> baseAccepter = baseObj.getAccepter();
+      expect(baseAccepter(Platypus.new1()), true);
+
+      ObjCBlock<Bool Function(Mammal)> childAccepter = childObj.getAccepter();
+      expect(childAccepter(Mammal.new1()), false);
+      expect(childAccepter(Platypus.new1()), true);
+
+      final platypus = Platypus.new1();
+      ObjCBlock<Platypus Function()> platypusReturner =
+          ObjCBlock_Platypus.fromFunction(() => platypus);
+      expect(baseObj.invokeReturner_(platypusReturner).laysEggs(), true);
+
+      final mammal = Mammal.new1();
+      ObjCBlock<Mammal Function()> mammalReturner =
+          ObjCBlock_Mammal.fromFunction(() => mammal);
+      expect(childObj.invokeReturner_(mammalReturner).laysEggs(), false);
+      expect(childObj.invokeReturner_(platypusReturner).laysEggs(), true);
+
+      ObjCBlock<Bool Function(Mammal)> mammalAccepter =
+          ObjCBlock_bool_Mammal.fromFunction(
+              (Mammal mammal) => mammal.laysEggs());
+      expect(baseObj.invokeAccepter_(mammalAccepter), true);
+
+      ObjCBlock<Bool Function(Platypus)> platypusAccepter =
+          ObjCBlock_bool_Platypus.fromFunction(
```

Comment:
nit: the signature uses the C type `Bool`, but the class usses `bool` Dart type.

Reply:
Surprisingly hard to fix. At the point where we're generating the block name, we don't have access to a `Writer` object, which is necessary to call all the `Type`'s canonical string methods like `getObjCBlockSignatureType`.